### PR TITLE
Set up CI pipeline for continuous integration

### PR DESCRIPTION
**Pull request by Awambeng**
- Created a CI pipeline to automate testing and building.
- Configured required status checks for the CI pipeline to ensure code quality and consistency before merging pull requests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "build": "npm run build --workspaces",
     "format": "prettier --write .",
     "format:check": "prettier --check .",
-    "lint": "eslint packages/** --ext .ts",
-    "lint:fix": "eslint --fix packages/** --ext .ts"
+    "lint": "eslint packages/** --ext .ts || echo 'No matching files'",
+    "lint:fix": "eslint --fix packages/** --ext .ts || echo 'No matching files'"
```
Comment (IngridPuppet) [inline package.json:11]: Could you further investigate a way to not fail when there are no matching files? The current construct will never fail in the future, even when it should.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+
+        with:
```
Comment (IngridPuppet) [inline .github/workflows/ci.yml:61]: Remove the extra line.
```suggestion
        uses: actions/setup-node@v4
        with:
```

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

**Comment by Awambeng** [comment]
@IngridPuppet i adjusted the changes requested

File: package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
     "build": "npm run build --workspaces",
     "format": "prettier --write .",
     "format:check": "prettier --check .",
-    "lint": "eslint packages/** --ext .ts",
-    "lint:fix": "eslint --fix packages/** --ext .ts"
+    "lint": "find packages/ -name '*.ts' | grep -q . && eslint packages/** --ext .ts || echo 'No matching files'",
+    "lint:fix": "find packages/ -name '*.ts' | grep -q . && eslint --fix packages/** --ext .ts || echo 'No matching files'"
```
Comment (IngridPuppet) [inline package.json:11]: This is getting really complicated. Can you try with this alternative?
```bash
eslint packages/** --ext .ts --no-error-on-unmatched-pattern
```

Comment (Awambeng) [reply package.json:11]: okay

**Review by IngridPuppet — APPROVED** [review]

**Pull request merged**